### DISCUSSION TITLE
Comparison with other formats

### DISCUSSION THREAD

**Discussion by hchiba1** [pg-format/pg-formatter#40] [General]
Especially, YARS-PG
* https://github.com/lszeremeta/yarspg/blob/main/other-notations/YARSpg.ebnf
* https://github.com/lszeremeta/yarspg/tree/main/yarspg/examples

GraphML?

GraphSON?

GML?

**Comment by hchiba1** [comment]
YARS-PG grammar:
```ebnf
node_id   ::= ALNUMPLUS
ALNUMPLUS ::= [a-zA-Z_] [a-zA-Z0-9_]*
```

**Comment by nichtich** [comment]
See https://github.com/pg-format/pgraphs/blob/dev/docs/features.csv for a comparision of data model (not to be confused with syntax!) of graph formats. This turned out to become a full research project but I don't know when I'll have time to summarize a paper. The features table should be extended by a column with regex of allowed node identifiers (parts of this are already hidden in the source of pgraphs serializers).

**Reply by nichtich** [reply]
Added some documentation: https://github.com/pg-format/pgraphs/blob/dev/docs/features.md
